Add zipkin deployment option to helm

douglas-reid opened the pull request:
This PR restores the option to use [zipkin](https://zipkin.io/) proper to implement the `zipkin` service in Istio deployments.

No attempt is made to incorporate significant productionization options for Zipkin at this time. It uses a single in-memory store + UI backend for now.

I looked at using the helm templates in https://github.com/Financial-Times/zipkin-helm, however there were some environmental assumptions (aws), etc., in those templates that made their use non-ideal out-of-the-box. Hopefully, as this moves forward, we can find a fully-configurable option (or operator) to use as a replacement.

cc: @adriancole, @costinm 

Example in action:
![zipkin](https://user-images.githubusercontent.com/21148125/48373413-97ac9700-e676-11e8-8969-75d019d9f9d0.png)

istio-testing commented:
@douglas-reid: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests-cni.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/155) | `/test e2e-simpleTests-cni`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests-cni.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/155) | `/test e2e-simpleTests-cni`
prow/build-tests.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1381) | `/test build-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests-cni.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/155) | `/test e2e-simpleTests-cni`
prow/build-tests.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1381) | `/test build-tests`
prow/istio-unit-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16610) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-simpleTests-cni.sh | 80a5886e357c450e2fc7db6da3d41aeee0c31df5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/155) | `/test e2e-simpleTests-cni`
prow/istio-unit-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16610) | `/test istio-unit-tests`
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16610) | `/test istio-unit-tests`
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/e2e-simpleTests-cni.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/156) | `/test e2e-simpleTests-cni`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

douglas-reid commented:
/test istio-unit-tests

objectiser commented on install/kubernetes/helm/subcharts/tracing/values.yaml:
```diff
@@ -22,7 +22,19 @@ jaeger:
       #     - jaeger.local
-replicaCount: 1
+zipkin:
+  hub: docker.io/openzipkin/zipkin
```
hub seems to be used in other sub-charts for the org, rather than the image - so maybe change to `docker.io/openzipkin` and add the `zipkin` image name into the deployment-zipkin.yaml.

douglas-reid replied:
done.

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/e2e-simpleTests-cni.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/156) | `/test e2e-simpleTests-cni`
prow/istio-pilot-multicluster-e2e.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4992) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

objectiser commented on install/kubernetes/helm/subcharts/tracing/templates/deployment-zipkin.yaml:
```diff
@@ -0,0 +1,57 @@
+              path: /health
+              port: {{ .Values.zipkin.queryPort }}
+          resources:
+            limits:
```
Resources should use the same approach as in other deployments, e.g. [prometheus](https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/prometheus/templates/deployment.yaml#L49-L53), [jaeger](https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/prometheus/templates/deployment.yaml#L49-L53), ...

Same applies for [priorityClassName, imagePullSecrets](https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/tracing/templates/deployment.yaml#L27-L35) and [affinity](https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/tracing/templates/deployment.yaml#L77-L78).

douglas-reid replied:
Done. I used separate `zipkin` and `jaeger` resource blocks, however, as I feel like they may not need the same default resources and I want to ease switching between the two.

objectiser commented on install/kubernetes/helm/subcharts/tracing/templates/deployment-zipkin.yaml:
```diff
@@ -0,0 +1,57 @@
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.zipkin.hub }}:{{ .Values.zipkin.tag }}"
```
As mentioned elsewhere, might be more consistent to have hub represent the org, and add image `{{ .Values.zipkin.hub }}/zipkin:{{ .Values.zipkin.tag }}`

douglas-reid replied:
done.

douglas-reid commented:
@objectiser thanks for the review! PTAL and let me know what you think.

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-pilot-multicluster-e2e.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4992) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/158) | `/test e2e-simpleTests-cni`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3/11062) | `/test istio-pilot-e2e-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-pilot-multicluster-e2e.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4992) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/158) | `/test e2e-simpleTests-cni`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3/11062) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-integ-k8s-tests.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/628) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/e2e-simpleTests-cni.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/158) | `/test e2e-simpleTests-cni`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3/11062) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-integ-k8s-tests.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/628) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4998) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

objectiser commented on install/kubernetes/helm/subcharts/tracing/values.yaml:
```diff
@@ -22,12 +24,30 @@ jaeger:
       # - secretName: jaeger-tls
       #   hosts:
       #     - jaeger.local
-replicaCount: 1
```
This is used in `deployment-jaeger.yaml`. Probably not relevant for in-memory config, so may be best to remove from the jaeger deployment aswell.

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-integ-k8s-tests.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/628) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4998) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/161) | `/test e2e-simpleTests-cni`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

adriancole commented:
@jonathan-lo @musaprg @kahootali @mclarke47 can one of you review this? maybe kick the tires? you'll be a better judge than me. Meanwhile, I'll take a look for things I might detect.

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-pilot-multicluster-e2e.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4998) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/161) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/633) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-pilot-multicluster-e2e.sh | da8ce1d3a4449e0f68bd6046db89a5f1ade693a0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/4998) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/161) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/633) | `/test istio-integ-k8s-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3-mcp.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3-mcp/2193) | `/test istio-pilot-e2e-envoyv2-v1alpha3-mcp`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/e2e-simpleTests-cni.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/161) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/633) | `/test istio-integ-k8s-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3-mcp.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3-mcp/2193) | `/test istio-pilot-e2e-envoyv2-v1alpha3-mcp`
prow/istio-pilot-multicluster-e2e.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5004) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

adriancole commented on install/kubernetes/helm/subcharts/tracing/templates/deployment-zipkin.yaml:
```diff
@@ -0,0 +1,56 @@
+              value: "-XX:ConcGCThreads={{ .Values.zipkin.node.cpus }} -XX:ParallelGCThreads={{ .Values.zipkin.node.cpus }} -Djava.util.concurrent.ForkJoinPool.common.parallelism={{ .Values.zipkin.node.cpus }} -Xms{{ .Values.zipkin.javaOptsHeap }}M -Xmx{{ .Values.zipkin.javaOptsHeap }}M -XX:+UseG1GC -server"
+            - name: COLLECTOR_SAMPLE_RATE
+              value: "{{ .Values.zipkin.sampleRate }}"
+            - name: QUERY_ENABLED
```
I'd remove this as it isn't common to change this, and having this could be considered distracting.

douglas-reid replied:
Done.

adriancole commented on install/kubernetes/helm/subcharts/tracing/templates/deployment-zipkin.yaml:
```diff
@@ -0,0 +1,56 @@
+              value: "{{ .Values.zipkin.queryPort }}"
+            - name: JAVA_OPTS
+              value: "-XX:ConcGCThreads={{ .Values.zipkin.node.cpus }} -XX:ParallelGCThreads={{ .Values.zipkin.node.cpus }} -Djava.util.concurrent.ForkJoinPool.common.parallelism={{ .Values.zipkin.node.cpus }} -Xms{{ .Values.zipkin.javaOptsHeap }}M -Xmx{{ .Values.zipkin.javaOptsHeap }}M -XX:+UseG1GC -server"
+            - name: COLLECTOR_SAMPLE_RATE
```
I'd remove this as collector-tier sampling is not great. If you are looking to limit the memory used, you can set `ZIPKIN_STORAGE_MEM_MAXSPANS=500000` This is the most frequently asked about thing for people testing as it allows them to limit memory even if not strictly (we don't control the size, rather count, and some instrumentation especially opentracing can create vastly different size spans depending on who wrote what)

https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/main/resources/zipkin-server-shared.yml#L55

douglas-reid replied:
Done.

objectiser commented on install/kubernetes/helm/subcharts/tracing/templates/deployment-zipkin.yaml:
```diff
@@ -0,0 +1,56 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-zipkin
```
Missing the namespace.

douglas-reid replied:
Good catch. Done.

mclarke47 commented:
@adriancole Looks similar to what I had in https://github.com/Financial-Times/zipkin-helm 👍

objectiser commented:
@douglas-reid The fix for the k8s ingress is https://github.com/istio/istio/pull/9930.

douglas-reid commented:
@adriancole thanks for the review. I believe I've addressed your comments now.  And yes, a separate PR will be required for the docs repo. It might have to follow the resolution of the in-flight PR around tracing, but it will be done after this PR is merged.

Please let me know if this PR is something we should prioritize for merging over to the `release-1.1` branch.

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/e2e-simpleTests-cni.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/161) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/633) | `/test istio-integ-k8s-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3-mcp.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3-mcp/2193) | `/test istio-pilot-e2e-envoyv2-v1alpha3-mcp`
prow/istio-pilot-multicluster-e2e.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5004) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16656) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3-mcp.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-e2e-envoyv2-v1alpha3-mcp/2193) | `/test istio-pilot-e2e-envoyv2-v1alpha3-mcp`
prow/istio-pilot-multicluster-e2e.sh | 7b4585d7a8625b5f352ba0d03d81df5986be882d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5004) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16656) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/175) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/664) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-unit-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-unit-tests/16656) | `/test istio-unit-tests`
prow/e2e-simpleTests-cni.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/175) | `/test e2e-simpleTests-cni`
prow/istio-integ-k8s-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/664) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5022) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

douglas-reid commented:
/test istio-unit-tests

mandarjog commented:
/lgtm

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/9910#" title="Author self-approved">douglas-reid</a>*, *<a href="https://github.com/istio/istio/pull/9910#issuecomment-438741265" title="LGTM">mandarjog</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[install/OWNERS](https://github.com/istio/istio/blob/master/install/OWNERS)~~ [mandarjog]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-integ-k8s-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/664) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5022) | `/test istio-pilot-multicluster-e2e`
prow/e2e-simpleTests-cni.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/193) | `/test e2e-simpleTests-cni`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/build-tests.sh | e8f59c41326b40b6f86146620658a519953909ed | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/build-tests/1383) | `/test build-tests`
prow/istio-integ-k8s-tests.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-integ-k8s-tests/664) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests-cni.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/e2e-simpleTests-cni/193) | `/test e2e-simpleTests-cni`
prow/istio-pilot-multicluster-e2e.sh | 31ced64911c89b806235eb7ba0a7ea4dc2445090 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/9910/istio-pilot-multicluster-e2e/5065) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

adriancole commented:
Still haven't seen the PR to restore the docs.. maybe holidays interfered?

adriancole commented:
ah great. thanks!

On Thu, 29 Nov 2018, 12:27 Douglas Reid <notifications@github.com wrote:

> @adriancole <https://github.com/adriancole> istio/istio.io#2884
> <https://github.com/istio/istio.io/pull/2884>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/9910#issuecomment-442670510>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAD613nRvl1rDwR2Yq8szaEx8eWAjCI5ks5uzzftgaJpZM4YaX_W>
> .
>

hklai commented:
@douglas-reid @adriancole @mclarke47 @objectiser @mandarjog 
it looks like this commit was merged to master but is missing from release-1.1. Is this needed in release-1.1?

douglas-reid commented:
@hklai i believe a separate PR added this to release-1.1 and I see the changes in that branch, so I think we are good.
